pkg/controller/jobs/deployment/deployment_webhook.go

```diff
@@ -52,21 +52,23 @@ func SetupWebhook(mgr ctrl.Manager, opts ...jobframework.Option) error {
 		Complete()
 }
 
-// +kubebuilder:webhook:path=/mutate-apps-v1-deployment,mutating=true,failurePolicy=fail,sideEffects=None,groups="apps",resources=deployments,verbs=create,versions=v1,name=mdeployment.kb.io,admissionReviewVersions=v1
+// +kubebuilder:webhook:path=/mutate-apps-v1-deployment,mutating=true,failurePolicy=fail,sideEffects=None,groups="apps",resources=deployments,verbs=create;update,versions=v1,name=mdeployment.kb.io,admissionReviewVersions=v1
 
 var _ admission.CustomDefaulter = &Webhook{}
 
 func (wh *Webhook) Default(ctx context.Context, obj runtime.Object) error {
-	d := fromObject(obj)
-	log := ctrl.LoggerFrom(ctx).WithName("deployment-webhook").WithValues("deployment", klog.KObj(d))
+	deployment := fromObject(obj)
+
+	log := ctrl.LoggerFrom(ctx).WithName("deployment-webhook").WithValues("deployment", klog.KObj(deployment))
 	log.V(5).Info("Applying defaults")
 
-	cqLabel, ok := d.Labels[constants.QueueLabel]
-	if ok {
-		if d.Spec.Template.Labels == nil {
-			d.Spec.Template.Labels = make(map[string]string, 1)
+	if queueName := jobframework.QueueNameForObject(deployment.Object()); queueName != "" {
+		if deployment.Spec.Template.Labels == nil {
+			deployment.Spec.Template.Labels = make(map[string]string, 1)
 		}
-		d.Spec.Template.Labels[constants.QueueLabel] = cqLabel
+		deployment.Spec.Template.Labels[constants.QueueLabel] = queueName
+	} else if wh.manageJobsWithoutQueueName {
+		delete(deployment.Spec.Template.Labels, constants.QueueLabel)
```

Comment:
@mimowo can we merge https://github.com/kubernetes-sigs/kueue/pull/3315 to allow delete on WebHook?

Reply:
Or we need to hold for https://github.com/kubernetes-sigs/controller-runtime/pull/2982.